topo_processor/metadata/lds_cache/lds_cache.py

```diff
@@ -51,11 +53,31 @@ def get_metadata(
 
             if os.path.isfile(metadata_path):
                 if exported_asset.extra_fields.get("encoding", None) == "gzip":
-                    decompress_file(metadata_path)
+                    if is_csv(metadata_path):
+                        decompress_file(metadata_path, False)
+                    elif is_geopackage(metadata_path):
+                        decompress_file(metadata_path, True)
+                        new_metadata_path = os.path.splitext(metadata_path)[0] + ".csv"
+                        geopackage_to_csv(metadata_path, new_metadata_path).run()
+                        metadata_path = new_metadata_path
+                    else:
+                        raise Exception(f"Unsupported file format. {metadata_path} must be .csv or .gpkg")
             else:
                 raise Exception(f"{metadata_path} not found")
 
     if os.path.isfile(metadata_path):
+        # This doesn't handle gzipped files as input on the command line
+        if is_geopackage(metadata_path):
+            new_metadata_path = os.path.splitext(metadata_path)[0] + ".csv"
+            if os.path.isfile(new_metadata_path):
+                raise Exception(
+                    f"Conversion of gpkg file {metadata_path} will overwrite existing csv file {new_metadata_path}"
```

Comment:
Did you want to stop the process at this point? I have the feeling that the message is a warning message rather than an error. I also don't see it as a problem to overwrite it as it is generated from the LDS cache.